### PR TITLE
DISCO-1388 course run redirect component

### PR DESCRIPTION
**Pull request by xnick421**

### CODE REVIEW

File: src/data/services/DiscoveryDataApiService.js [hunk 1]
```diff
@@ -17,6 +17,19 @@ class DiscoveryDataApiService {
     });
   }
 
+  static fetchCourseRuns(params) {
```
Comment (Dillon-Dumesnil) [inline src/data/services/DiscoveryDataApiService.js]: This feels like a weird function name when it's returning a single course_uuid. You aren't fetching any course runs. Maybe `fetchCourseThroughCourseRun` or something like that, but `fetchCourseRuns` feels weird.

Comment (Dillon-Dumesnil) [reply src/data/services/DiscoveryDataApiService.js]: Also shouldn't this more closely follow the above method? We are only sending in a single course run, right? Why not make the path `${DiscoveryDataApiService.discoveryBaseUrl}/course_runs/${key}`?

File: src/components/CourseRunRedirect/index.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        .fetchCourseRuns({ keys: this.props.courseRunKey });
+      const uuid = response.data.results[0].course_uuid;
+      this.setState({ path: `/courses/${uuid}` }); // eslint-disable-line react/no-did-mount-set-state
+    } catch (e) {
+      this.setState({ path: '/' }); // eslint-disable-line react/no-did-mount-set-state
+    }
+  }
+
+  render() {
+    return this.state.path ? <Redirect to={this.state.path} /> : null;
```
Comment (Dillon-Dumesnil) [inline src/components/CourseRunRedirect/index.jsx]: This probably shouldn't be rendering null and instead render the `LoadingSpinner` component so it is consistent with actually loading a course.

File: src/data/services/DiscoveryDataApiService.js (hunk 1 above)
Comment (xnick421) [reply src/data/services/DiscoveryDataApiService.js]: Your totally right, didn't realize the course run detail accepted a key

**Pull request merged**